packages/api/cms-api/src/dam/files/files.controller.ts

```diff
@@ -9,136 +9,163 @@ import {
-    }
+export function createFilesController({ Scope: PassedScope }: { Scope?: Type<DamScopeInterface> }): Type<unknown> {
+    const Scope = PassedScope ?? EmptyDamScope;
+    const hasNonEmptyScope = PassedScope != null;
 
-    @Get(`/preview/${fileUrl}`)
-    async previewFileUrl(@Param() params: FileParams, @Res() res: Response, @Headers("range") range?: string): Promise<void> {
-        return this.streamFile(params, res, { range, overrideHeaders: { "Cache-control": "private" } });
+    function nonEmptyScopeOrNothing(scope: DamScopeInterface): DamScopeInterface | undefined {
+        return hasNonEmptyScope ? scope : undefined;
     }
 
-    @DisableGlobalGuard()
-    @Get(`/:hash/${fileUrl}`)
-    async hashedFileUrl(
-        @Param() { hash, ...params }: HashFileParams,
-        @Res() res: Response,
-        @Headers(CDN_ORIGIN_CHECK_HEADER) cdnOriginCheck: string,
-        @Headers("range") range?: string,
-    ): Promise<void> {
-        this.checkCdnOrigin(cdnOriginCheck);
-
-        if (!this.isValidHash(hash, params)) {
-            throw new NotFoundException();
-        }
+    const UploadFileBody = createUploadFileBody({ Scope });
 
-        return this.streamFile(params, res, { range });
-    }
+    @Controller("dam/files")
+    class FilesController {
+        constructor(
+            @Inject(DAM_CONFIG) private readonly damConfig: DamConfig,
+            private readonly filesService: FilesService,
+            private readonly blobStorageBackendService: BlobStorageBackendService,
+        ) {}
 
-    private checkCdnOrigin(incomingCdnOriginHeader: string): void {
-        if (this.damConfig.cdnEnabled) {
-            if (incomingCdnOriginHeader !== this.damConfig.cdnOriginHeader) {
-                throw new ForbiddenException();
+        @Post("upload")
+        @UseInterceptors(DamUploadFileInterceptor(FilesService.UPLOAD_FIELD))
+        async upload(@UploadedFile() file: FileUploadInterface, @Body() body: UploadFileBodyInterface): Promise<FileInterface> {
+            const transformedBody = plainToInstance(UploadFileBody, body);
+            const errors = await validate(transformedBody, { whitelist: true, forbidNonWhitelisted: true });
+
+            if (errors.length > 0) {
+                throw new CometValidationException("Validation failed", errors);
             }
+
+            const { scope, folderId } = transformedBody;
+            const uploadedFile = await this.filesService.upload({ file, folderId }, nonEmptyScopeOrNothing(scope));
+            return Object.assign(uploadedFile, { fileUrl: await this.filesService.createFileUrl(uploadedFile) });
         }
-    }
 
-    private isValidHash(hash: string, fileParams: FileParams): boolean {
-        return hash === this.filesService.createHash(fileParams);
-    }
+        @Get(`/preview/${fileUrl}`)
+        async previewFileUrl(@Param() params: FileParams, @Res() res: Response, @Headers("range") range?: string): Promise<void> {
```

Comment:
I think you need to check scope here

Reply:
same in images controller

Reply:
Done: 22817404de5423ecf9c7f9bcea20e1e917ca0e5b